### PR TITLE
[FEATURE] Add plugin.tx_solr.index.enableCommits option.

### PR DESCRIPTION
**Pull request by RasmusWernerLarsen**
The option makes it possible to disable implicit solr commits, which
can improve scalability and performance when a solr core is shared
between many installations.

This affects the garbageCollector (used by the indexQueue) and the special reIndexTask.

This doesn't preclude manually triggering commits and the deleteByType function will still issue commits by default, but that can be explicitly disabled, and changing that would cause unnecessary API breakage.

I still need to test this a bit more, so please don't merge it yet.

Fixes: #870

### CODE REVIEW

**Comment by timohund** [comment]
@RasmusWernerLarsen can you please apply php-cs-fixer the build has shown some violations.

File: Classes/GarbageCollector.php [hunk 1]
```diff
@@ -201,12 +201,15 @@ protected function deleteIndexDocuments($table, $uid)
...
         foreach ($indexQueueItems as $indexQueueItem) {
             $site = $indexQueueItem->getSite();
+            $solrConfiguration = $site->getSolrConfiguration();
 
             // a site can have multiple connections (cores / languages)
             $solrConnections = $connectionManager->getConnectionsBySite($site);
             foreach ($solrConnections as $solr) {
                 $solr->deleteByQuery('type:' . $table . ' AND uid:' . intval($uid));
-                $solr->commit(false, false, false);
+                if ($solrConfiguration->getEnableCommits()) {
```
Comment (timohund) [inline Classes/GarbageCollector.php]: I think, retrieving the setting could also be done outside the loop. What do you think?

Comment (RasmusWernerLarsen) [inline Classes/GarbageCollector.php]: Good point 👍

**Comment by timohund** [comment]
Looks good to me, thanks for the pull request!

**Pull request merged**